src/authentication/service/password.auth.service.ts

```diff
@@ -56,6 +62,62 @@ export default class PasswordAuthService implements Authenticatable {
     return this.userService.createUser(userFromInput);
   }
 
+  async inviteTokenSignup(
+    userDetails: UserInviteTokenSignupInput,
+  ): Promise<InviteTokenResponse> {
+    const verifyUser = await this.userService.verifyDuplicateUser(
+      userDetails.email,
+      userDetails.phone,
+    );
+    if (verifyUser.existingUserDetails) {
+      throw new UserExistsException(
+        verifyUser.existingUserDetails,
+        verifyUser.duplicate,
+      );
+    }
+    const userFromInput = new User();
+    userFromInput.email = userDetails.email;
+    userFromInput.phone = userDetails.phone;
+    userFromInput.firstName = userDetails.firstName;
+    userFromInput.middleName = userDetails.middleName;
+    userFromInput.lastName = userDetails.lastName;
+    userFromInput.status = Status.INVITED;
+    const savedUser = await this.userService.createUser(userFromInput);
+    const invitationToken = this.authenticationHelper.generateInvitationToken(
+      { id: savedUser.id },
+      '7d',
+    );
+    await this.userService.updateField(
+      savedUser.id,
+      'inviteToken',
+      invitationToken.token,
+    );
```

Comment:
Should we first create a user and then update the token or should we create a new user with the token making the operation atomic?
- How do we address the scenario when user creation without password was successful but creation and updating invite token failed.